src/rulesets/LinearAlgebra/norm.jl

```diff
@@ -17,20 +17,25 @@ function frule((_, Δx), ::typeof(norm), x::Number, p::Real)
     return y, ∂y
 end
 
-function rrule(
-    ::typeof(norm),
-    x::Union{StridedArray, LinearAlgebra.AbstractTriangular, Diagonal},
-    p::Real,
-)
+function rrule(::typeof(norm), x::AbstractArray, p::Real)
     y = LinearAlgebra.norm(x, p)
-    function norm_pullback(Δy)
+    function norm_pullback_p(Δy)
         ∂x = Thunk() do
             return if isempty(x) || p == 0
-                zero.(x) .* (zero(y) * zero(real(Δy)))
+                InplaceableThunk(
+                    @thunk(zero.(x) .* (zero(y) * zero(real(Δy)))),
```

Comment:
You know what is fun? this is broken for some arrays.
We might want to constrain the `eltype` to be `Number`

Consider:

```
julia> norm([[1],[2]], 2.0)
2.23606797749979

julia> zero(eltype([[1],[2]]))
ERROR: MethodError: no method matching zero(::Type{Vector{Int64}})
```

Reply:
Not blocking for this PR

Reply:
This entire file should probably be restricted to arrays of `<:Number`?